scripts/tasks/generate-license-files.js

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2024, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+const path = require('node:path');
+const { readFile, writeFile, stat, readdir } = require('node:fs/promises');
+const prettier = require('prettier');
+const { BUNDLED_DEPENDENCIES } = require('../shared/bundled-dependencies.js');
+
+// Generate our LICENSE files for each package, including any bundled dependencies
+// This is modeled after how Rollup does it:
+// https://github.com/rollup/rollup/blob/0b665c3/build-plugins/generate-license-file.ts
+
+// Async fs.existsSync
+async function exists(filename) {
+    try {
+        await stat(filename);
+        return true;
+    } catch (err) {
+        return false;
+    }
+}
+
+async function findLicenseText(depName) {
+    // Iterate through possible names for the license file
+    const names = ['LICENSE', 'LICENSE.md', 'LICENSE.txt'];
```

Comment:
If you want to be robust, the license file could match any `/^licen[cs]e(?:\..*?)$/i`, or the complex syntax allowed in the package.json `license` field.

Reply:
Good point! All our existing deps just have a `LICENSE*` file.

BTW I opened a PR on `@parse5/tools` to actually add a `LICENSE` file.